Complete Printing Cleanup

jolyonb opened the pull request:
This PR applies a number of changes to the print media CSS.

Being able to print nicely from edX is important for a number of reasons, including making PDFs for students to view offline, and for a variety of accessibility issues. Currently, printing in the edX platform is unusable. For example, here is a print to pdf of a page from MITx: http://web.mit.edu/jolyon/Public/printing_present.pdf

This pull request modifies a number of things in the print media CSS to make this beautiful. I'll jump straight to the punchline. Here is what that same page looks like with this pull request in place: http://web.mit.edu/jolyon/Public/printing_with_changes.pdf

(Two other pull requests are also in place for this second pdf; https://github.com/edx/edx-platform/pull/7592 and https://github.com/edx/edx-platform/pull/7591)

All of the changes included in this PR modify print media CSS *only*. I'll go through and annotate each change individually, so that you can see what it is doing. Most of the changes are hiding things; a few of them are changing borders and whitespace.

edx-webhook commented:
Thanks for the pull request, @jolyonb! I've created [OSPR-502](https://openedx.atlassian.net/browse/OSPR-502) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams. 

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ('this must be merged by XX date', and why that is)
- partner information ('this is a course on edx.org')
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the Github pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

We can't start reviewing your pull request until you've submitted a [signed contributor agreement](http://open.edx.org/sites/default/files/wysiwyg/individual-contributor-agreement.pdf) or indicated your institutional affiliation and added yourself to the [AUTHORS](https://github.com/edx/edx-platform/blob/master/AUTHORS) file. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst) file for more information.

jolyonb commented on common/lib/xmodule/xmodule/css/capa/display.scss:
```diff
@@ -269,6 +269,13 @@ div.problem {
       }
     }
 
+    // Hides equation previews in symbolic response problems when printing
```
One feature of symbolic inputs is a preview for students, as shown below. This change hides the preview from printing.
![screen shot 2015-04-03 at 11 21 58 pm](https://cloud.githubusercontent.com/assets/6232546/6991274/505436a2-da58-11e4-8ae1-360621fec93e.png)

jolyonb commented on common/lib/xmodule/xmodule/css/capa/display.scss:
```diff
@@ -655,6 +662,10 @@ div.problem {
       -webkit-font-smoothing: antialiased;
     }
+
+    @media print {
```
This change hides all problem action buttons in the printing. Buttons include "Save", "Check", "Show/Hide Answer", etc inside problems.

sarina replied:
Will this hide how many checks are remaining, or other information that appear after the button? It's reasonable to hide the buttons but other information should be preserved.

jolyonb replied:
This presently hides the remaining checks information as well. It should be reasonably straightforward to be more specific to hide only buttons, but leave the check information.

jolyonb commented on common/lib/xmodule/xmodule/css/sequence/display.scss:
```diff
@@ -235,6 +235,10 @@ $sequence--border-color: #C8C8C8;
     display: none;
   }
+
+  @media print {
```
This hides the sequential navigation bar.

sarina replied:
I don't like this. It removes information from the page, particularly, where in the sequence this page was printed from. This can become very important for sequences that have a very large number of units. Is there a way to just print this information better? `@media print` rules can be used in many varied and dynamic ways, not just to show or hide things.

jolyonb replied:
It's already hiding information about the chapter and sequence names that the page is printing from. I'd love to display all of it in the header, but I've no idea how I'd put it there without changing the page templates. Given that the present "sequential bar" method of navigation completely obscures all information that is present in each tab already, I don't know a good way to fix this. If the tabs were numbered on the screen, this would be a completely different story.

jolyonb commented on common/lib/xmodule/xmodule/css/sequence/display.scss:
```diff
@@ -302,6 +306,10 @@ nav.sequence-bottom {
   margin: lh(2) auto;
   text-align: center;
+
+  @media print {
```
This hides the navigation left/right arrows at the bottom of a vertical.

jolyonb commented on lms/static/sass/base/_base.scss:
```diff
@@ -122,6 +122,10 @@ a:focus {
   background: $content-wrapper-bg;
   padding-bottom: ($baseline*2);
+
+  @media print {
```
This removes 40px of whitespace from the bottom of the page. This whitespace looks good on screen, but can force a new page to be printed (even if it has nothing on it).

jolyonb commented on lms/static/sass/base/_base.scss:
```diff
@@ -267,6 +271,10 @@ mark {
     }
   }
+
+  @media print {
```
This stops the "Help" tab on the left side of the screen from being printed.

jolyonb commented on lms/static/sass/base/_base.scss:
```diff
@@ -353,4 +361,8 @@ mark {
     height: auto;
   }
+
+  @media print {
```
This ensures that the "Skip to content" navigational element is not printed.

jolyonb commented on lms/static/sass/base/_reset.scss:
```diff
@@ -96,4 +96,5 @@ td { vertical-align: top; }
   @page { margin: 0.5cm; }
   p, h2, h3 { orphans: 3; widows: 3; }
   h2, h3 { page-break-after: avoid; }
+  .donotprint { display:none !important; }
```
Sometimes, it is useful to have a method of making sure that something in your course does not print (for example, if it only really has meaning in an interactive webpage). This defines a class called "donotprint" that makes sure that that element does not print, but shows normally on the screen.

sarina replied:
I am against this entirely. This could very easily be misused by course staff. If you really want stuff to not print in your course, you can inject that CSS directly. If we include this, we have two options:

1. We document this - thus implicitly endorsing hiding your content to those who may need to print for a11y reasons. I don't like this.
2. We don't document this - thus it's a secret that really only you know about. So, why not just ask you to inject this, really teeny amount of css, directly into your unit.

jolyonb replied:
Just being complete - there are CSS classes defined to hide things on the screen. Injecting print media CSS is a pain (you can't do it inline).

marcotuts replied:
A few notes here @jolyonb  - It looks like you're not using .donotprint here, and I agree with sarina that this seems an unnecessary workaround to include behind the scenes. 

nit / suggestion: That being said, in a similar vein, I would suggest adding an extend to _extends.scss (within lms/static/sass/base) called %ui-print-excluded (or something similar). That way, if we have any other print specific styles in the future, we can quickly separate the intent of the display: none rule.

jolyonb replied:
Indeed, I'm not using .donotprint here. That said, it's one line with zero side effects, it's function is obvious, and it will make implementation a whole lot easier for anybody who wants to use it. Implementing it yourself inside a course is really somewhat annoying.

Would you like me to go through the rest of the code and change all the display:none's to extend such a placeholder selector?

marcotuts replied:
1.) I'm in agreement with Sarina here that .donotprint should be removed prior to merge here. Authors can still use display: none if they would really like to exclude more additional special case content. 

2.) %ui-print-excluded { @media print { display: none} } 
I would add this to _extends.scss so that we can help make the styling easier to read and maintain. That way, any class you want to apply this to simply gets a @extend %ui-print-excluded rule. This also helps reduce cases where media queries are floating off far away from the class name they belong to (which is common in our old, very strangely nested css).

jolyonb commented on lms/static/sass/course/_info.scss:
```diff
@@ -229,5 +229,13 @@ div.info-wrapper {
       }
     }
+
+    @media print {
```
This ensures that the "Handouts" sidebar on the course info page does not print. Although it may contain useful information, it typically runs over the margin and off the page, and generally doesn't print nicely.

sarina replied:
I don't like this remediation. Change the print rule to make it print nicely, not to remove the information.

jolyonb replied:
I struggled to do this. I'll give it another go this weekend.

jolyonb commented on lms/static/sass/course/_info.scss:
```diff
@@ -229,5 +229,13 @@ div.info-wrapper {
+    }
+  }
+
+  @media print {
```
This ensures that an ugly grey border is not printed around the tab.

sarina replied:
I don't know what you mean by this. Screenshot please?

jolyonb replied:
See screenshot below:
![screen shot 2015-04-17 at 4 00 04 pm](https://cloud.githubusercontent.com/assets/6232546/7209900/fff57b1c-e51a-11e4-93e5-e7afa08701c3.png)
It's the same border for every other border you've asked for a screenshot for, just that it exists in different parts of the CSS.

jolyonb commented on lms/static/sass/course/_tabs.scss:
```diff
@@ -9,4 +9,8 @@ div.static_tab_wrapper {
     margin: 0px 0px 20px;
   }
+
+  @media print {
```
This ensures that an ugly grey border is not printed around a course tab.

sarina replied:
I don't know what you mean by this. Screenshot please?

jolyonb replied:
See screenshot below:
![screen shot 2015-04-17 at 4 00 04 pm](https://cloud.githubusercontent.com/assets/6232546/7209900/fff57b1c-e51a-11e4-93e5-e7afa08701c3.png)
It's the same border for every other border you've asked for a screenshot for, just that it exists in different parts of the CSS. (Note: this comment was previously on a previous line, that got eaten when it became an old diff)

jolyonb commented on lms/static/sass/course/courseware/_courseware.scss:
```diff
@@ -37,6 +37,10 @@ html.video-fullscreen{
     }
   }
+
+  @media print {
```
This ensures that instructor buttons such as "View in Studio", "Staff Debug Info", and "Student Submission History" do not print.

sarina replied:
I don't love this rule, because it obscures completely that you've printed from a staff user account.

jolyonb replied:
I see printing as being useful from two perspectives: creating PDF versions of a page for students to download, and allowing students to print. If I'm making a PDF version, I don't particularly want "Staff Info Only" plastered all over it. I'd much prefer to create something that is as close to what students will see as possible.

jolyonb commented on lms/static/sass/course/layout/_courseware_header.scss:
```diff
@@ -59,6 +59,10 @@
     }
   }
+
+  @media print {
```
This ensures that the navigational bar at the top of the page (with Courseware, etc) doesn't print.

jolyonb commented on lms/static/sass/course/modules/_calculator.scss:
```diff
@@ -177,4 +177,8 @@ div.calc-main {
     }
   }
+
+  @media print {
```
This ensures that the calculator doesn't print.

jolyonb commented on lms/static/sass/discussion/_discussion.scss:
```diff
@@ -1003,6 +1003,10 @@ body.discussion {
     }
   }
+
+  @media print {
```
This ensures that inline discussion boards don't print.

jolyonb commented on lms/static/sass/shared/_footer.scss:
```diff
@@ -190,6 +190,10 @@
     }
   }
+
+  @media print {
```
This ensures that the footer doesn't print. The edX footer is really quite large, and it would be a waste of ink and paper to print that every time!

sarina replied:
I agree that the whole footer shouldn't print, but something out to print. I'm thinking something like a boxed plaintext message that indicates that "This page was printed from {SITE_NAME}. (c) 2015 {PLATFORM_NAME}"

jolyonb replied:
That would be neat, but would require changing the page template rather than a minimal CSS change. The header still includes the platform logo, platform name, and site name (see image below).

marcotuts replied:
@jolyonb I would echo the suggested change to only hide a part of the footer instead of the whole thing. By scoping your rule changes you can make the edx.org footer printable.

.footer-about --> add print media query  that makes it span 12 columns instead of 6, and .footer-nav plus .footer-follow would get the display none extend. You could also then go further down and hide the longer about us text section, plus the links to Terms of Service, Privacy Policy, and Accessibility Policy links for the print view as well. 

It is a few more rules but you shouldn't need to edit the template.

jolyonb replied:
If you just want to get down to a copyright notice, then that's straightforward. When Sarina wanted the course name as well, that required digging into the template.

I'll make some changes to keep a small footer present.

jolyonb commented on lms/static/sass/shared/_header.scss:
```diff
@@ -264,6 +264,10 @@ header.global {
       }
     }
+
+    @media print {
```
This ensures that the user dropdown box in the header doesn't print.

sarina replied:
But the rest of the header prints?

jolyonb replied:
Yes, see the PDF example I linked in the introduction to this PR:
![screen shot 2015-04-17 at 3 38 29 pm](https://cloud.githubusercontent.com/assets/6232546/7209577/e31a2928-e517-11e4-8457-0f97449e8419.png)

jolyonb commented on lms/static/sass/course/courseware/_courseware.scss:
```diff
@@ -240,6 +244,10 @@ div.course-wrapper {
       }
     }
+
+    @media print {
```
Removes extra whitespace from around the content. The page that's being printed on already has a margin.

jolyonb commented on lms/static/sass/course/courseware/_courseware.scss:
```diff
@@ -274,6 +282,10 @@ div.course-wrapper {
   }
+
+  @media print {
+    border: 0;
```
Removes a border that normally prints around the course content, but doesn't look good when printed (especially since the right hand side of the border is usually off the page).

jolyonb commented:
Notification: @carsongee

sarina commented:
@jolyonb - I'm confused - you have about 8 pull requests open at this point. Is this PR a summary of all those other pull requests, or additional to it, or what? This is a lot of work and I'm not following what is going on here.

jolyonb commented:
HI @sarina, sorry for the confusion. Each PR here is a separate unit in its own right, although four of them have to do with printing in the LMS. I tried to divide the PRs into logical chunks:

* Fix page margin issue
* Fix link issue
* Fix background image issue
* General fix of what is being printed and what's not.

This PR is just the final point here, and is by far the largest change; I couldn't figure out a way to divide this into any smaller chunks. The smaller chunks each split off changes to _reset.scss, and associated knock-on effects. Each PR is completely independent.

Each of the individual changes made here are either hiding something (I tried to screenshot everything that's being hidden for clarity), or changing margins/fixing borders.

jolyonb commented:
I'll add two things this weekend:

* Keep the number of checks remaining information at the bottom of a problem.
* Try to keep the course handout sidebar on the page.

jolyonb commented on common/lib/xmodule/xmodule/css/capa/display.scss:
```diff
@@ -628,6 +635,10 @@ div.problem {
       font-weight: 600;
+
+      @media print {
+        display: none;
```
Hides the Show/Check/Reset/Save buttons; keeps any submission feedback text.

marcotuts replied:
I can understand some of the other removed elements (borders, details, tangential navigation controls, etc) but at some point removing elements like the input buttons presumes some level of use / intent around the printed document. Is the expectation here that users printing these pages expect the problems to not reflect their input controls? It also seems a bit odd that checkbox controls would remain unchanged but multiple choice problems would have their buttons stripped. Thoughts? (@jolyonb)

jolyonb replied:
This only hides the buttons down the bottom of a problem to save/check/etc. Multiple choices still have their little round buttons, checkboxes are still present, and whatever the state the user presently has is what prints. The buttons to actually submit an answer and elicit a response from the server are all that I've stripped here.

For the purpose of having a hard copy of a page, be it for readability, the ability to scribble on the page, to just have a PDF record, or to journal/archive, the buttons down the bottom really don't add anything to the result, and can look misplaced to boot (especially a button that says "Hide Answers"). I honestly don't think that much if anything will be lost by hiding these buttons.

marcotuts replied:
Thanks for the note here, that makes sense to me. As you noted these input controls don't change the content / setup of the problem.

jolyonb commented:
So, it turns out that the Course Handouts box on the Welcome splashpage extends the sidebar class, which already has a print media command to hide all sidebars from printing. I've taken out my code that simply duplicated this feature.

@sarina had requested that the course handouts be kept in place with better CSS, rather than just hiding them. However, I think this would require modifying the sidebar class, which is extended by a number of other classes, and I'm not game to do that at this stage.

sarina commented:
ok to test

sarina commented:
Thanks for your responses. Our UX team is really slammed with reviews of a bunch of open source and internal work, but they will get to this as soon as they are able to.

jolyonb commented:
@marcotuts, I've changed the footer around. The CSS there is a mess of old stuff and new stuff, insofar as I can tell. I've just left a copyright message down the bottom, as the edX logo is already printing at the top of the first page.

I can't actually test this; I only have access to an old open edx version that is running off the old footer, so you probably want to test it out.

jolyonb commented:
I've removed the .donotprint line. I'll just reiterate one last time, you cannot use inline CSS to specify print media styling.

marcotuts commented:
test this please

marcotuts commented:
@jolyonb  - Tests are kicked off again for this, and I'll be spinning up a sandbox for testing this morning. I'm still seeing a handful of cases where we could be using the %ui-print-excluded extend instead of directly adding display: none;  Finally, aside from any other items that may come up in sandbox testing, I would remove the "!important" flag on the extend rule.  

Thanks!

jolyonb commented:
@marcotuts - Everything is changed over to an extends command. Should be ready to go, I think?

marcotuts commented:
@jolyonb  - Quick update here. I just want to test these changes on a sandbox, which I plan to spin up tomorrow morning. In particular I want to double check the footer styling for the various cases, since you noted wanting some help testing all the cases for that. Thanks for making those suggested changes!

jolyonb commented:
@marcotuts I think I've successfully rebased this? I had some git issues, so not entirely sure (one day I'll learn git properly). Let me know how it goes.

If I may also point this out to you: there is one more pull request I have open with regards to getting printing fixed, located here: 
https://github.com/edx/edx-platform/pull/7589
It fixes a problem with green checkmarks and red crosses not printing.

sarina commented:
@jolyonb pull requests are prioritized into team sprints, which last two
weeks. Unfortunately due to other commitments of work, we cannot generally
get things reviewed on the fly unless they are urgent. #7589 will be
reviewed when the team is able to have space in an upcoming sprint for the
work. Thanks for understanding.

On Wed, May 13, 2015 at 12:06 AM, jolyonb <notifications@github.com> wrote:

> @marcotuts <https://github.com/marcotuts> I think I've successfully
> rebased this? I had some git issues, so not entirely sure (one day I'll
> learn git properly). Let me know how it goes.
>
> If I may also point this out to you: there is one more pull request I have
> open with regards to getting printing fixed, located here:
> #7589 <https://github.com/edx/edx-platform/pull/7589>
> It fixes a problem with green checkmarks and red crosses not printing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/7595#issuecomment-101508004>.
>

marcotuts commented:
@jolyonb - I'll try and rebuild the sandbox using this new rebased PR. I ran into issues trying to provision a sandbox earlier this morning on the PR before it was rebased. I am still seeing 2 failing tests (down from 50!) so I'll try and kick off tests again to see if it is a flaky test. (It seems to be a Youtube test, so it isn't related to this PR as far as I can tell.) 

Thanks for flagging that other PR, but as Sarina noted, this will go through the standard OSPR process. I may not wind up being the reviewer at that point, it depends. 

test this please

marcotuts commented:
test this please

sarina commented:
@marcotuts can you compile the assets on this branch locally? the builds are failing because of a css issue: https://build.testeng.edx.org/job/edx-platform-test-subset/58270/console

jolyonb commented:
I see the issue in the log, and know exactly what line is causing it. However, I haven't the foggiest idea why that line is causing it. I'll revert that line to a media print display none instead of extending the other class, and we'll see if that fixes things.

jolyonb commented:
@marcotuts @sarina We'll see if that line fixes the tests?

marcotuts commented:
@jolyonb  - I realized from that exchange what is going on. I would revert your extends in xmodule/css/sequence/display.scss

See line 36, where @cpennington  notes that extends don't work for capa. That should hopefully fix it. Thanks @sarina  for the help diagnosing that.

jolyonb commented:
@marcotuts I don't know what exchange you're talking about, nor which line 36 you refer to. Regardless, I've changed all of my extends in the xmodule/css/capa/display.scss. Do you want me to do likewise for xmodule/css/sequence/display.scss?

marcotuts commented:
@jolyonb  - Ah sorry, by exchange I meant your and sarina's comments. I saw your latest edit so it looks like we're good to go. Tests should hopefully pass this time around.

marcotuts commented on lms/static/sass/shared/_footer.scss:
```diff
@@ -190,6 +190,11 @@
+
+  @media print {
+    border: none;
+    padding: 10px 10px 0 10px;
```
I just reviewed this locally, and it isn't clear to me what the border: 0 is resetting here. Additionally I would keep the default padding of 25x 10px 0 10px; eliminating the need for this media query in its entirety. 

If I'm wrong and the border is in fact resetting something, please move this up to line 12 so the media query is more clearly associated with its parent / container.

jolyonb replied:
Part of this was trying to deal more globally with openedx sites using the old footer vs the new footer. It may well have been extraneous.

marcotuts commented on lms/static/sass/shared/_footer.scss:
```diff
@@ -358,9 +374,14 @@ $edx-footer-bg-color: rgb(252,252,252);
       margin-bottom: ($edx-footer-spacing/2);
     }
+
+    @media print {
```
in my testing here, adding float: left to this area ensured that the print footer copyright text and the openedx logo is presented in the same line instead of being stacked. 

This is what I'm seeing without the float: left rule: 
![image](https://cloud.githubusercontent.com/assets/2023680/7619318/db505a96-f985-11e4-89f5-101c7a791734.png)

jolyonb replied:
Ok. I tried adding this in via an include statement; not sure what it looks like at the end of the day. You'll want to test.

marcotuts commented on lms/static/sass/shared/_footer.scss:
```diff
@@ -488,4 +510,9 @@ $edx-footer-bg-color: rgb(252,252,252);
     }
   }
+
+  @media print {
```
this should also be moved to the top of the list of styles being added to this level of depth in the SASS, making it clearer what its parent container is.

jolyonb replied:
Done.

sarina commented on lms/static/sass/shared/_footer.scss:
```diff
@@ -301,13 +306,23 @@ $edx-footer-bg-color: rgb(252,252,252);
+      width: 100%;
+      border: none;
+      padding-right: none;
+      margin-right: none;
```
@marcotuts should rules like this use the 
```css
@include margin-right(none);
```

syntax to account for RTL?

marcotuts replied:
Ah yeah, I commented on that inline as well, but github collapsed the change: https://github.com/edx/edx-platform/pull/7595#discussion_r30268028

We just want to make sure it is margin-right(0); instead of "none" but +1

marcotuts commented on lms/static/sass/shared/_footer.scss:
```diff
@@ -301,13 +306,23 @@ $edx-footer-bg-color: rgb(252,252,252);
     width: flex-grid(6,12);
+
+    @media print {
+      width: 100%;
```
NIT: width: flex-grid (12,12);

marcotuts commented on lms/static/sass/shared/_footer.scss:
```diff
@@ -301,13 +306,23 @@ $edx-footer-bg-color: rgb(252,252,252);
+    @media print {
+      width: 100%;
+      border: none;
+      padding-right: none;
```
Change this rule and the margin rule afterward to use 0 instead of none. This issue resulted in some strange rendering for me where the courseware was much narrower than intended. 

Also, for RTL support use the following:
@include margin-right(0); @include padding-right(0);

marcotuts commented on lms/static/sass/shared/_footer.scss:
```diff
@@ -488,4 +510,9 @@ $edx-footer-bg-color: rgb(252,252,252);
+
+  @media print {
+    border: none;
+    padding: 10px 10px 0 10px;
```
For RTL support, use @include padding(x, x, x, x);

marcotuts commented:
@jolyonb - A couple other small changes came up when I did my local devstack testing, including some small RTL support changes.

jolyonb commented:
@marcotuts I'll go and fix these this evening. There still seem to be failed checks; do you know what these are about?

sarina commented:
@jolyonb when you do your fixes, do another rebase (there were some lingering errors) and we can suss out any remaining failures tomorrow.

jolyonb commented:
@sarina @marcotuts Fixed footer things, squashed and rebased. Again, you'll want to test the footer stuff, as I can't. Hopefully the failing tests will be fixed this time too.

jolyonb commented:
Well, the old tests that were failing now pass. However, there's a new test that was passing that now fails. Again, doesn't look like it has anything to do with these changes.

sarina commented:
test this please

jolyonb commented:
Wha??? 275 failures? Perhaps God doesn't want us to print... ;-)

sarina commented:
test this please

sarina commented:
@jolyonb Test "common.test.acceptance.tests.lms.test_learner_profile.DifferentUserLearnerProfilePageTest.test_different_user_public_profile" in bok choy did fail on both builds. Please try running locally to see if your changes are causing a failure, it's not reported as a flaky test. https://github.com/edx/edx-platform/blob/master/docs/en_us/internal/testing.rst

jolyonb commented:
@sarina Looks like it passed all the tests this time. Do you still want me to try a local test cycle?

sarina commented:
@jolyonb no, I will file a flaky test report, no need to test locally.

@marcotuts got a passing build. Are there more review need to be done?

marcotuts commented:
:+1:  from me! the issued I uncovered in my last test locally have been resolved. thanks @jolyonb  and @sarina

jolyonb commented:
Yay! Celebrations! :fireworks:

jolyonb commented:
@marcotuts @sarina So, it looks like somebody completely redid the edX footer, and all of the print styling is off again. Shall I have another go at bashing the print media CSS for the footer into shape, or should I leave that for somebody else?

sarina commented:
@jolyonb I don't know if anyone here will have the bandwidth to tackle that. @AlasdairSwan I believe redid the footer, perhaps he can do some fixes - I recommend syncing up with him.
